cpp/src/arrow/compute/exec/exec_plan.h

```diff
@@ -419,37 +429,116 @@ struct ARROW_EXPORT Declaration {
 /// This method will add a sink node to the declaration to collect results into a
 /// table.  It will then create an ExecPlan from the declaration, start the exec plan,
 /// block until the plan has finished, and return the created table.
-ARROW_EXPORT Result<std::shared_ptr<Table>> DeclarationToTable(
-    Declaration declaration, ExecContext* exec_context = default_exec_context());
+///
+/// If `use_threads` is false then all CPU work will be done on the calling thread.  I/O
+/// tasks will still happen on the I/O executor and may be multi-threaded (but should
+/// not use significant CPU resources)
+ARROW_EXPORT Result<std::shared_ptr<Table>> DeclarationToTable(Declaration declaration,
+                                                               bool use_threads = true);
 
 /// \brief Asynchronous version of \see DeclarationToTable
+///
+/// The behavior of use_threads is slightly different than the synchronous version since
+/// we cannot run synchronously on the calling thread.  Instead, if use_threads=false then
+/// a new thread pool will be created with a single thread and this will be used for all
+/// compute work.
+///
+/// If a custom exec context is provided then the value of `use_threads` will be ignored
+/// and the executor in the custom context will be used.  However, the executor must
```

Comment:
It doesn't seem to be possible to pass both an exec context and use_threads anymore.

Reply:
Ah, good catch.  I went through several iterations of the API and this comment only made sense somewhere in the middle.